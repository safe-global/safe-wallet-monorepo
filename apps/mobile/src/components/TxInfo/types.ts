import { Transaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { TxType } from '@/src/hooks/useTransactionType'

export type TTxCardPress = { tx: Transaction; type: TxType }
