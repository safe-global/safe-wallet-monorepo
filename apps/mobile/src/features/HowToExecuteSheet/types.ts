export enum ExecutionMethod {
  WITH_RELAY = 'with_relay',
  WITH_PK = 'with_pk',
}
