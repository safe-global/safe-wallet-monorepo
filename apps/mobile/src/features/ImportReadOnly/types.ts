import { z } from 'zod'
import { formSchema } from '@/src/features/ImportReadOnly/schema'

export type FormValues = z.infer<typeof formSchema>
