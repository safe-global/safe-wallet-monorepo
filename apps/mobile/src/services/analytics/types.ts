import type {
  Transaction,
  TransferTransactionInfo,
  SettingsChangeTransaction,
} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
export type {
  Transaction,
  TransferTransactionInfo,
  SettingsChangeTransaction,
  CustomTransactionInfo,
  SwapTransferTransactionInfo,
} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import type { ANALYTICS_LABELS } from './constants'

/**
 * Firebase Analytics Event Types
 * These event names are passed directly to Firebase Analytics
 */
export enum EventType {
  SCREEN_VIEW = 'screen_view',
  CLICK = 'customClick',
  META = 'metadata',
  SAFE_APP = 'safeApp',
  SAFE_CREATED = 'safe_created',
  SAFE_ACTIVATED = 'safe_activated',
  SAFE_OPENED = 'safe_opened',
  WALLET_CONNECTED = 'wallet_connected',
  TX_CREATED = 'tx_created',
  TX_CONFIRMED = 'tx_confirmed',
  TX_EXECUTED = 'tx_executed',
}

export type EventLabel = string | number | boolean | null

export type AnalyticsEvent = {
  eventName: string
  eventCategory: string
  eventAction: string
  eventLabel?: EventLabel
  chainId?: string
}

export enum AnalyticsUserProperties {
  WALLET_LABEL = 'walletLabel',
  WALLET_ADDRESS = 'walletAddress',
}

// Extract precise types from the source of truth
export type TransactionInfoType = Transaction['txInfo']['type']
export type TransferInfoType = TransferTransactionInfo['transferInfo']['type']
export type SettingsInfoType = SettingsChangeTransaction['settingsInfo']['type']

// Union of all possible analytics labels
export type AnalyticsLabel =
  | (typeof ANALYTICS_LABELS.BASE_TYPES)[TransactionInfoType]
  | (typeof ANALYTICS_LABELS.TRANSFER_TYPES)[TransferInfoType]
  | (typeof ANALYTICS_LABELS.SETTINGS_TYPES)[SettingsInfoType]
  | (typeof ANALYTICS_LABELS.ENHANCED)[keyof typeof ANALYTICS_LABELS.ENHANCED]

/**
 * MixPanel User Attributes for Safe Wallet (Mobile)
 *
 * These attributes are used for cohort analysis and user segmentation
 * Shared with web app for consistency
 */
export interface SafeUserAttributes {
  /** On-chain address of the Safe (Primary key for cohort analysis) */
  safe_id: string

  /** Timestamp when the Safe was deployed (Life of user) */
  created_at: Date

  /** Each version has different properties (Filter out older Safes) */
  safe_version: string

  /** Total number of wallet addresses/signers on this Safe (Engagement metric) */
  num_signers: number

  /** Number of signatures required to execute a transaction (Setup intent) */
  threshold: number

  /** Blockchain networks (Context for segmentation) */
  networks: string[]

  /** Timestamp of most recent transaction (Churn alerting) */
  last_tx_at: Date | null

  /** ID of parent Space if the Safe is grouped in a Space (Space-level grouping) */
  space_id: string | null

  /** Array of child Safe IDs that this Safe contains (Nested safes grouping) */
  nested_safe_ids: string[]

  /** Lifetime number of transactions executed by this Safe (Engagement metric) */
  total_tx_count: number
}

/**
 * Event tracking types for MixPanel with humanized property names
 */
export interface SafeEventProperties {
  'Safe ID': string
  'Safe Version': string
  Network: string
  [key: string]: unknown
}
