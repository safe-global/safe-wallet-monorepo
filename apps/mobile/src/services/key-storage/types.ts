export type PrivateKeyStorageOptions = {
  requireAuthentication?: boolean
}

export interface IKeyStorageService {
  storePrivateKey(userId: string, privateKey: string, options?: PrivateKeyStorageOptions): Promise<void>
  getPrivateKey(userId: string, options?: PrivateKeyStorageOptions): Promise<string | undefined>
  removePrivateKey(userId: string, options?: PrivateKeyStorageOptions): Promise<void>
}
