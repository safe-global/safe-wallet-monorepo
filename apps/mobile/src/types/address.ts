export interface SafeInfo {
  address: Address
  chainId: string
}

export type Address = `0x${string}`
