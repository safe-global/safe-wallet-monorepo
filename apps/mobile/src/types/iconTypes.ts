export type IconName =
  | 'add-owner'
  | 'address-book'
  | 'alert-triangle'
  | 'alert'
  | 'appearance'
  | 'apps'
  | 'arrow-down'
  | 'arrow-left'
  | 'arrow-sort'
  | 'arrow-up'
  | 'arrow-right'
  | 'batch'
  | 'bell'
  | 'block'
  | 'blocks-1'
  | 'blocks'
  | 'bookmark-filled'
  | 'bookmark'
  | 'camera'
  | 'chat'
  | 'check-filled'
  | 'check-oulined'
  | 'check'
  | 'chevron-down'
  | 'chevron-left'
  | 'chevron-right'
  | 'chevron-up'
  | 'clock'
  | 'close-filled'
  | 'close-outlined'
  | 'close'
  | 'code-blocks'
  | 'copy'
  | 'dapp-logo'
  | 'delete'
  | 'desktop'
  | 'document'
  | 'double-arrow'
  | 'download'
  | 'dropdown-arrow-small'
  | 'edit-owner'
  | 'edit'
  | 'element-drag'
  | 'experimental'
  | 'export'
  | 'external-link'
  | 'eye-n'
  | 'eye-off'
  | 'file'
  | 'filter'
  | 'hardware'
  | 'home'
  | 'info'
  | 'key'
  | 'keystone'
  | 'ledger'
  | 'lightbulb'
  | 'link'
  | 'lock'
  | 'mobile'
  | 'nft'
  | 'options-horizontal'
  | 'options-vertical'
  | 'owners'
  | 'paste'
  | 'plus-filled'
  | 'plus-outlined'
  | 'plus'
  | 'points'
  | 'qr-code-1'
  | 'qr-code'
  | 'question'
  | 'repeat'
  | 'replace-owner'
  | 'rows-1'
  | 'rows-2'
  | 'rows'
  | 'safe'
  | 'scan-1'
  | 'scan'
  | 'search'
  | 'seed'
  | 'send-to'
  | 'settings'
  | 'share'
  | 'shield-crossed'
  | 'shield'
  | 'sign'
  | 'tag'
  | 'token'
  | 'transaction-batch'
  | 'transaction-change-settings'
  | 'transaction-contract'
  | 'transaction-execute'
  | 'transaction-incoming'
  | 'transaction-outgoing'
  | 'transaction-partial-fill'
  | 'transaction-recovery'
  | 'transaction-stake'
  | 'transaction-swap'
  | 'transactions'
  | 'unlock'
  | 'update'
  | 'upload'
  | 'wallet'
  | 'what-is-new'

export const iconNames: IconName[] = [
  'add-owner',
  'address-book',
  'alert-triangle',
  'alert',
  'appearance',
  'apps',
  'arrow-down',
  'arrow-left',
  'arrow-sort',
  'arrow-up',
  'arrow-right',
  'batch',
  'bell',
  'block',
  'blocks-1',
  'blocks',
  'bookmark-filled',
  'bookmark',
  'camera',
  'chat',
  'check-filled',
  'check-oulined',
  'check',
  'chevron-down',
  'chevron-left',
  'chevron-right',
  'chevron-up',
  'clock',
  'close-filled',
  'close-outlined',
  'close',
  'code-blocks',
  'copy',
  'dapp-logo',
  'delete',
  'desktop',
  'document',
  'double-arrow',
  'download',
  'dropdown-arrow-small',
  'edit-owner',
  'edit',
  'element-drag',
  'experimental',
  'export',
  'external-link',
  'eye-n',
  'eye-off',
  'file',
  'filter',
  'hardware',
  'home',
  'info',
  'key',
  'keystone',
  'ledger',
  'lightbulb',
  'link',
  'lock',
  'mobile',
  'nft',
  'options-horizontal',
  'options-vertical',
  'owners',
  'paste',
  'plus-filled',
  'plus-outlined',
  'plus',
  'points',
  'qr-code-1',
  'qr-code',
  'question',
  'repeat',
  'replace-owner',
  'rows-1',
  'rows-2',
  'rows',
  'safe',
  'scan-1',
  'scan',
  'search',
  'seed',
  'send-to',
  'settings',
  'share',
  'shield-crossed',
  'shield',
  'sign',
  'tag',
  'token',
  'transaction-batch',
  'transaction-change-settings',
  'transaction-contract',
  'transaction-execute',
  'transaction-incoming',
  'transaction-outgoing',
  'transaction-partial-fill',
  'transaction-recovery',
  'transaction-stake',
  'transaction-swap',
  'transactions',
  'unlock',
  'update',
  'upload',
  'wallet',
  'what-is-new',
]
