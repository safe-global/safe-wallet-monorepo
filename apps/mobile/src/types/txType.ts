export enum ETxType {
  ADD_SIGNER = 'ADD_SIGNER',
  REMOVE_SIGNER = 'REMOVE_SIGNER',
  STAKE = 'STAKE',
  CONTRACT_INTERACTION = 'CONTRACT_INTERACTION',
  SWAP_ORDER = 'SWAP_ORDER',
  NFT_TRANSFER = 'NFT_TRANSFER',
  TOKEN_TRANSFER = 'TOKEN_TRANSFER',
}
