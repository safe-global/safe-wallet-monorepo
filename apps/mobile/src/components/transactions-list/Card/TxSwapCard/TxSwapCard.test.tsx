import { render } from '@/src/tests/test-utils'
import { TxSwapCard } from '.'
import { OrderTransactionInfo } from '@safe-global/store/gateway/types'
import { mockSwapTransfer } from '@/src/tests/mocks'

describe('TxSwapCard', () => {
  it('should render the default markup', () => {
    const container = render(<TxSwapCard onPress={() => null} txInfo={mockSwapTransfer as OrderTransactionInfo} />)

    expect(container).toMatchSnapshot()
  })
})
