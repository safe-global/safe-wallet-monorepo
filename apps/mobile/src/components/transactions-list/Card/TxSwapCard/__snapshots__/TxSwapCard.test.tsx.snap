// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxSwapCard should render the default markup 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F4F4F4",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 12,
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 12,
          "maxWidth": "55%",
        }
      }
    >
      <View
        style={
          {
            "height": 40,
            "position": "relative",
            "width": 40,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 100000,
              "borderBottomRightRadius": 100000,
              "borderTopLeftRadius": 100000,
              "borderTopRightRadius": 100000,
              "flexDirection": "column",
              "height": 28,
              "justifyContent": "center",
              "maxHeight": 28,
              "maxWidth": 28,
              "minHeight": 28,
              "minWidth": 28,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "position": "absolute",
              "top": 0,
              "width": 28,
            }
          }
        >
          <View
            style={
              {
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
          >
            <Image
              accessibilityLabel="SafeToken"
              fullscreen={true}
              onError={[Function]}
              onLoad={[Function]}
              source={
                {
                  "uri": "https://safe-transaction-assets.safe.global/tokens/logos/0x5aFE3855358E112B5647B952709E6165e1c1eEEe.png",
                }
              }
              style={
                {
                  "backgroundColor": "#EEEFF0",
                  "height": 28,
                  "width": 28,
                }
              }
            />
          </View>
          <View
            style={
              {
                "backgroundColor": "#EEEFF0",
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#121312",
              "borderBottomLeftRadius": 100000,
              "borderBottomRightRadius": 100000,
              "borderTopLeftRadius": 100000,
              "borderTopRightRadius": 100000,
              "bottom": 0,
              "flexDirection": "column",
              "height": 28,
              "justifyContent": "center",
              "maxHeight": 28,
              "maxWidth": 28,
              "minHeight": 28,
              "minWidth": 28,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "position": "absolute",
              "right": 0,
              "width": 28,
            }
          }
        >
          <View
            style={
              {
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
          >
            <Image
              accessibilityLabel="Ether"
              fullscreen={true}
              onError={[Function]}
              onLoad={[Function]}
              source={
                {
                  "uri": "https://safe-transaction-assets.safe.global/chains/1/chain_logo.png",
                }
              }
              style={
                {
                  "backgroundColor": "#EEEFF0",
                  "height": 28,
                  "width": 28,
                }
              }
            />
          </View>
          <View
            style={
              {
                "backgroundColor": "#EEEFF0",
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          />
        </View>
      </View>
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 4,
              "marginBottom": 4,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#A1A3A7",
                  "fontSize": 10,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="safe-list-transaction-swap-icon"
          >
            
          </Text>
          <Text
            style={
              {
                "color": "#A1A3A7",
                "fontSize": 12,
                "marginBottom": 2,
              }
            }
            suppressHighlighting={true}
          >
            Swap order
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#121312",
              "fontSize": 16,
              "fontWeight": 600,
            }
          }
          suppressHighlighting={true}
        >
          SAFE &gt; ETH
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        style={
          {
            "color": "#121312",
          }
        }
        suppressHighlighting={true}
      >
        +
        0.05
         
        ETH
      </Text>
      <Text
        style={
          {
            "color": "#121312",
            "fontSize": 12,
          }
        }
        suppressHighlighting={true}
      >
        −
        0.05
         
        SAFE
      </Text>
    </View>
  </View>
</View>
`;
